ui/apps/platform/src/Containers/Dashboard/PatternFly/Widgets/AgingImages.tsx

```diff
@@ -0,0 +1,252 @@
+import React, { useReducer } from 'react';
+import {
+    Flex,
+    FlexItem,
+    Title,
+    Button,
+    Dropdown,
+    DropdownToggle,
+    Form,
+    FormGroup,
+    Checkbox,
+    TextInput,
+    ValidatedOptions,
+} from '@patternfly/react-core';
+import { useQuery, gql } from '@apollo/client';
+import cloneDeep from 'lodash/cloneDeep';
+import pluralize from 'pluralize';
+
+import LinkShim from 'Components/PatternFly/LinkShim';
+import useURLSearch from 'hooks/useURLSearch';
+import { getRequestQueryStringForSearchFilter } from 'utils/searchUtils';
+import useSelectToggle from 'hooks/patternfly/useSelectToggle';
+import { SearchFilter } from 'types/search';
+import { vulnManagementImagesPath } from 'routePaths';
+import { getQueryString } from 'utils/queryStringUtils';
+import WidgetCard from './WidgetCard';
+import AgingImagesChart, {
+    TimeRangeCounts,
+    TimeRangeTupleIndex,
+    TimeRangeTuple,
+    timeRangeTupleIndices,
+} from './AgingImagesChart';
+
+export const imageCountQuery = gql`
+    query agingImagesQuery($query0: String, $query1: String, $query2: String, $query3: String) {
+        timeRange0: imageCount(query: $query0)
+        timeRange1: imageCount(query: $query1)
+        timeRange2: imageCount(query: $query2)
+        timeRange3: imageCount(query: $query3)
+    }
+`;
+
+function queryStringFor(timeRangeValue: number, searchFilter: SearchFilter) {
+    return getRequestQueryStringForSearchFilter({
+        ...searchFilter,
+        'Image Created Time': `>${timeRangeValue}d`,
+    });
+}
+
+type QueryVariables = Record<`query${TimeRangeTupleIndex}`, string>;
+
+function getQueryVariables(timeRanges: TimeRangeTuple, searchFilter: SearchFilter): QueryVariables {
+    return {
+        query0: queryStringFor(timeRanges[0].value, searchFilter),
+        query1: queryStringFor(timeRanges[1].value, searchFilter),
+        query2: queryStringFor(timeRanges[2].value, searchFilter),
+        query3: queryStringFor(timeRanges[3].value, searchFilter),
+    };
+}
+
+// Gets the header string title for the widget based on applied filters and resulting counts
+function getWidgetTitle(
+    searchFilter: SearchFilter,
+    selectedTimeRanges: TimeRangeTuple,
+    timeRangeCounts?: TimeRangeCounts
+): string {
+    if (!timeRangeCounts) {
+        return 'Aging images';
+    }
+
+    const totalImages =
+        Object.values(timeRangeCounts).find((range, index) => {
+            return selectedTimeRanges[index].enabled;
+        }) ?? 0;
+
+    const isActiveImages = Boolean(searchFilter.Cluster) || Boolean(searchFilter['Namespace ID']);
```

Comment:
What do you think about pure helper function as future refactoring after initial release?

Reply:
I think this is already handled in a downstream PR, as the pattern had emerged quite a few times across widgets.